src/modules/Auth/services/auth.ts

```diff
@@ -1,26 +1,81 @@
-export interface User {
-  name: string;
-  email: string;
-  password?: string;
+import Keycloak, { KeycloakConfig } from 'keycloak-js';
+
+const onLoad: Keycloak.KeycloakOnLoad = 'login-required';
+
+class Config implements KeycloakConfig {
+  url: string = process.env.AUTH_URL || 'auth';
+  realm: string = process.env.AUTH_REALM || 'sume';
+  clientId: string = process.env.AUTH_CLIENT_ID || 'sume-app';
 }
 
-export function SignIn(): Promise<any> {
-  return new Promise((resolve) =>
-    setTimeout(
-      () =>
-        resolve({
-          token: 'fake-token',
-          user: { name: 'Ricardo Lüders', email: 'my@email.com' },
-        }),
-      1000,
-    ),
-  );
+const _kc = Keycloak(new Config());
+
+function Init(appEntrypoint: () => void): void {
+  _kc
+    .init({
+      onLoad,
+    })
+    .then((auth) => {
+      if (!auth) {
+        window.location.reload();
```

Comment:
if auth fails this will redirect to keycloak login page?
have u tested if not will be stuck and reloading in sequence?

Reply:
Yeap! I've tested it. If the login fails it stays at Keycloaks UI until a valid authentication is made.